### PR TITLE
Fix bug in regexp logic for company name

### PR DESCRIPTION
**Pull request by autotelik**
Simplify regexp matching
Resolves bug where a single special char raised invalid (even though that's a valid name)

### CODE REVIEW

**Comment by autotelik** [comment]
:eyes:

File: app/validators/ea/validators/companies_house_name_validator.rb [hunk 1]
```diff
@@ -11,8 +11,10 @@ module Validators
 
     class CompaniesHouseNameValidator < ActiveModel::EachValidator
 
+      VALID_COMPANIES_HOUSE_NAME_REGEX = Regexp.new(/[\^|_~¬`]/).freeze
```
Comment (reggieb) [inline app/validators/ea/validators/companies_house_name_validator.rb]: I don't think you need the `Regexp.new`. 

```
/[\^|_~¬`]/.freeze 
```

should work.

File: app/validators/ea/validators/companies_house_name_validator.rb [hunk 2]
```diff
@@ -24,8 +26,8 @@ def self.name_max_length
         160
       end
 
-      def self.valid_name_regex
-        /\A[\p{Alpha}\p{N}][^\^|_~¬`]+\z/i
+      def self.disallowed_chars
+        @disallowed_chars ||= ["^", "|", "_", "~", "¬", "`"]
```
Comment (reggieb) [inline app/validators/ea/validators/companies_house_name_validator.rb]: I'd suggest this is a little clearer: 
```
@disallowed_chars ||= %w( ^ | _ ~ ¬ ` )
```

File: spec/validators/companies_house_name_validator_spec.rb [hunk 3]
```diff
@@ -14,33 +14,39 @@ class NameValidatable
...
+      end
+
+      context "With only a single special char" do
+        let(:name) { "@" }
+        it "is valid" do end
+      end
+
+      context "With only multiple special chara" do
+        let(:name) { "@ $$" }
+        it "is valid" do end
```
Comment (reggieb) [inline spec/validators/companies_house_name_validator_spec.rb]: The tests with `it "is valid" do end` aren't testing anything. If they are not needed, they should be removed.

Comment (reggieb) [inline spec/validators/companies_house_name_validator_spec.rb]: I think the use of an after block is confusing, as it forces the need for the strange `it "is valid" do end` entries. Would `is_expected.to be_valid` in each context be simpler?

**Comment by autotelik** [comment]
At this stage I'm not changing this code as its intentions are clear to me

**Pull request merged**